libswift/Sources/Optimizer/FunctionPasses/MergeCondFails.swift

```diff
@@ -0,0 +1,94 @@
+//===--- MergeCondFail.swift -  Merge cond_fail instructions --------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SIL
+
+let mergeCondFailsPass = FunctionPass(name: "merge-condfails", runMergeCondFails)
+
+/// Return true if the operand of the cond_fail instruction looks like
+/// the overflow bit of an arithmetic instruction.
+private func hasOverflowConditionOperand(_ cfi: CondFailInst) -> Bool {
+  if let tei = cfi.condition as? TupleExtractInst {
+    return tei.tuple is BuiltinInst
+  }
+  return false
+}
+
+/// Merge cond_fail instructions.
+///
+/// We can merge cond_fail instructions if there is no side-effect or memory
+/// write in between them.
+/// This pass merges cond_fail instructions by building the disjunction of
+/// their operands.
+private func runMergeCondFails(function: Function, context: FunctionPassContext) {
+
+  // Merge cond_fail instructions if there is no side-effect or read in
+  // between them.
+  for block in function.blocks {
+    // Per basic block list of cond_fails to merge.
+    var condFailToMerge = StackList<CondFailInst>(context)
+
+    for inst in block.instructions {
+      if let cfi = inst as? CondFailInst {
+        // Do not process arithmetic overflow checks. We typically generate more
+        // efficient code with separate jump-on-overflow.
+        if !hasOverflowConditionOperand(cfi) &&
+           (condFailToMerge.isEmpty || cfi.message == condFailToMerge.first!.message) {
+          condFailToMerge.push(cfi)
+        }
+      } else if inst.mayHaveSideEffects || inst.mayReadFromMemory {
+        // Stop merging at side-effects or reads from memory.
+        mergeCondFails(&condFailToMerge, context: context)
+      }
+    }
+    // Process any remaining cond_fail instructions in the current basic
+    // block.
+    mergeCondFails(&condFailToMerge, context: context)
+  }
+}
+
+/// Try to merge the cond_fail instructions. Returns true if any could
```

Comment:
This function doesn't have a return value

Reply:
Thanks, I'll fix it.